url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/term/Function.java

```diff
@@ -0,0 +1,689 @@
+            this.functions.put("cosh", new BuiltInFunction(Math.class.getMethod("cosh", double.class)));
+            this.functions.put("exp", new BuiltInFunction(Math.class.getMethod("exp", double.class)));
+            this.functions.put("fabs", new BuiltInFunction(Math.class.getMethod("abs", double.class)));
+            this.functions.put("floor", new BuiltInFunction(Math.class.getMethod("floor", double.class)));
+            this.functions.put("log", new BuiltInFunction(Math.class.getMethod("log", double.class)));
+            this.functions.put("log10", new BuiltInFunction(Math.class.getMethod("log10", double.class)));
+
+            this.functions.put("sin", new BuiltInFunction(Math.class.getMethod("sin", double.class)));
+            this.functions.put("sinh", new BuiltInFunction(Math.class.getMethod("sinh", double.class)));
+            this.functions.put("sqrt", new BuiltInFunction(Math.class.getMethod("sqrt", double.class)));
+            this.functions.put("tan", new BuiltInFunction(Math.class.getMethod("tan", double.class)));
+            this.functions.put("tanh", new BuiltInFunction(Math.class.getMethod("tanh", double.class)));
+
+            this.functions.put("log1p", new BuiltInFunction(Math.class.getMethod("log1p", double.class)));
+            //not found in Java
+//            this.functions.put("acosh", new BuiltInFunction("acosh",  & (acosh)));
+//            this.functions.put("asinh", new BuiltInFunction("asinh",  & (asinh)));
+//            this.functions.put("atanh", new BuiltInFunction("atanh",  & (atanh)));
+
+            this.functions.put("pow", new BuiltInFunction(Math.class.getMethod("pow", double.class, double.class)));
+            this.functions.put("atan2", new BuiltInFunction(Math.class.getMethod("atan2", double.class, double.class)));
+            this.functions.put("fmod", new BuiltInFunction(Op.class.getMethod("modulo", double.class, double.class)));
+        } catch (Exception ex) {
+            throw new RuntimeException("[function error] built-in functions could not be loaded", ex);
+        }
+    }
+
+    public boolean isOperator(String token) {
+        return this.operators.containsKey(token);
+    }
+
+    public boolean isBuiltInFunction(String token) {
+        return this.functions.containsKey(token);
+    }
+
+    public boolean isOperand(String token) {
+        //An operand is not a parenthesis...
+        if ("(".equals(token) || ")".equals(token) || ",".equals(token)) {
+            return false;
+        }
+        //nor an operator...
+        if (isOperator(token)) {
+            return false;
+        }
+        //nor a function...
+        if (isBuiltInFunction(token)) {
+            return false;
+        }
+        //...it is everything else :)
+        return true;
+    }
+
+    public String toPostfix(String formula) {
+        return toPostfix(formula, false);
+    }
+
+    public String toPostfix(final String formula, boolean loadFunctions) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 31. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 542-550 into a separate method.